plugins/cdvdGigaherz/src/CDVD.cpp

```diff
@@ -153,107 +142,81 @@ DWORD CALLBACK keepAliveThread(PVOID param)
     }
 
     printf(" * CDVD: KeepAlive thread finished.\n");
-
-    return 0;
 }
 
-s32 StartKeepAliveThread()
+bool StartKeepAliveThread()
 {
-    hNotify_keepAlive = CreateEvent(NULL, FALSE, FALSE, NULL);
-    if (hNotify_keepAlive == nullptr)
-        return -1;
-
     cdvdKeepAlive_is_open = true;
-    hThread_keepAlive = CreateThread(NULL, 0, keepAliveThread, NULL, 0, &pidThreadKeepAlive);
-    if (hThread_keepAlive == nullptr) {
+    try {
+        s_keepalive_thread = std::thread(keepAliveThread);
+    } catch (std::system_error &) {
         cdvdKeepAlive_is_open = false;
-        return -1;
     }
 
-    SetThreadPriority(hThread_keepAlive, THREAD_PRIORITY_NORMAL);
-
-    return 0;
+    return cdvdKeepAlive_is_open;
 }
 
 void StopKeepAliveThread()
 {
     cdvdKeepAlive_is_open = false;
-    PulseEvent(hNotify_keepAlive);
-    if (WaitForSingleObject(hThread_keepAlive, 5000) == WAIT_TIMEOUT) {
-        TerminateThread(hThread_keepAlive, 0);
-    }
-    CloseHandle(hThread_keepAlive);
-    CloseHandle(hNotify_keepAlive);
+    s_keepalive_cv.notify_one();
+    s_keepalive_thread.join();
 }
 
 ///////////////////////////////////////////////////////////////////////////////
 ///////////////////////////////////////////////////////////////////////////////
 // CDVD Pluin Interface                                                      //
 
-void CALLBACK CDVDsetSettingsDir(const char *dir)
+EXPORT void CALLBACK CDVDsetSettingsDir(const char *dir)
 {
-    CfgSetSettingsDir(dir);
+    s_config_file = std::string(dir ? dir : "inis") + "/cdvdGigaherz.ini";
 }
 
-s32 CALLBACK CDVDinit()
+EXPORT s32 CALLBACK CDVDinit()
 {
     return 0;
 }
 
-s32 CALLBACK CDVDopen(const char *pTitleFilename)
+EXPORT s32 CALLBACK CDVDopen(const char *pTitleFilename)
 {
     ReadSettings();
 
-    if (source_drive == '-') {
-        // MSDN : Trailing backslash is required to ensure consistent behavior across
-        // various versions of Windows and storage types.
-        char temp[] = "A:\\";
-
-        for (char d = 'A'; d <= 'Z'; d++) {
-            temp[0] = d;
-            if (GetDriveType(temp) == DRIVE_CDROM) {
-                source_drive = d;
-                break;
-            }
-        }
-        if (source_drive == '-')
-            return -1;
-    }
-
-    sprintf(csrc, "\\\\.\\%c:", source_drive);
-
-    printf(" * CDVD: Opening drive '%s'...\n", csrc);
+    auto drive = GetValidDrive();
+    if (drive.empty())
+        return -1;
 
     // open device file
-    src = new IOCtlSrc(csrc);
-
-    if (!src->IsOK()) {
-        printf(" * CDVD: Error opening source.\n");
+    try {
+        src = std::unique_ptr<IOCtlSrc>(new IOCtlSrc(drive));
+    } catch (std::runtime_error &ex) {
```

Comment:
Do you typically put `catch` on the same line? Looks somehow odd to me. But probably it is somehow consistent...